### PR TITLE
[ADD] sale_stock_available_info_popup: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
Adds  'Available to promise' quantity to qty_at_date_widget widget added in https://github.com/OCA/stock-logistics-warehouse/pull/887

Depends on:
- [x] https://github.com/OCA/stock-logistics-warehouse/pull/887

Cc @Tecnativa TT21547

### CODE REVIEW

File: sale_stock_available_info_popup/static/src/xml/qty_at_date.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2020 Tecnativa - Ernesto Tejeda
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<templates>
+    <div t-extend="sale_stock.QtyDetailPopOver">
+        <t t-jquery="t[t-if*='!data.is_mto']" t-operation="append">
+            <tr>
+                <td><strong>Available to promise</strong></td>
+                <td><t t-esc='data.immediately_usable_qty_today'/>
+                <t t-esc='data.product_uom.data.display_name'/></td>
```
Comment (Tardo) [inline sale_stock_available_info_popup/static/src/xml/qty_at_date.xml:15]: I can't test it... but is this correct?

**Comment by manuelcalerosolis** [comment]
There is no translations file in this module?

**Comment by pedrobaeza** [comment]
Please rebase

File: sale_stock_available_info_popup/static/src/xml/qty_at_date.xml (hunk 1 above)
Comment (ernestotejeda) [reply sale_stock_available_info_popup/static/src/xml/qty_at_date.xml:15]: Yes @Tardo It is in this way, because of the data structure in 'data'. Thanks.

**Comment by ernestotejeda** [comment]
@pedrobaeza Done

**Review by rousseldenis — APPROVED** [review]
Code + functional review. Works as expected

**Comment by pedrobaeza** [comment]
One question: the color of the info icon in which field is based? I think we should adjust the colors according this new field, or offer a configuration option.

**Comment by rousseldenis** [comment]
> One question: the color of the info icon in which field is based? I think we should adjust the colors according this new field, or offer a configuration option.

It depends on multiple fields and is based on classes : https://github.com/OCA/stock-logistics-warehouse/blob/12.0/sale_stock_info_popup/static/src/xml/qty_at_date.xml#L8

**Comment by pedrobaeza** [comment]
Yeah, and what do you think about changing its meaning when installing this module?

**Comment by rousseldenis** [comment]
> Yeah, and what do you think about changing its meaning when installing this module?

What do you mean ?

**Comment by pedrobaeza** [comment]
I mean that colors of the icon are set according "available to promise" field.

**Comment by ernestotejeda** [comment]
I agree with @pedrobaeza that the color depends on the quantity to promise instead of the forecasted quantity. Take a look at how I've done and tell me what do you think.

**Review by chienandalu — COMMENTED** [review]
Testing in runbot the popup doesn't pop

**Comment by pedrobaeza** [comment]
@chienandalu please re-review

**Review by chienandalu — APPROVED** [review]
Now it works right. Is it correct to not show stock qtys on consumables (I guess it is on purpose)

**Comment by pedrobaeza** [comment]
Yes, indeed consumables don't have stock control. Let's go then:

/ocabot merge nobump

**Pull request merged**